pkg/client/oci/oci.go

```diff
@@ -354,9 +405,113 @@ func (r *Repo) Reload() error {
 	return errors.Errorf("reload method is not supported yet")
 }
 
-func (r *Repo) newDockerResolver() remotes.Resolver {
+// SetEntries sets the entries of a repo
+func (r *Repo) SetEntries(entries map[string][]string) {
```

Comment:
It seems that you are creating this method just to enable testing from a different package `oci_test`. I personally don't like that. it usually means that we are mixing up unit testing with interface/functional testing. It also pollutes the package public interface with a method that's not meant to be used outside of testing.

Is there any other way to achieve the same effect without relying on this?